src/main/java/com/task/controller/CrudController.java

```diff
@@ -24,18 +24,29 @@ public class CrudController {
   @GetMapping("/names")
   public List<UserData> read() {
 
-    UserData[] UserData = new UserData[2];
-    UserData[0] = new UserData("tanaka", "1990/5/10");
-    UserData[1] = new UserData("suzuki", "2000/8/20");
+    UserData[] userData = new UserData[5];
+    userData[0] = new UserData("tanaka", "1990/05/10");
+    userData[1] = new UserData("suzuki", "2000/08/20");
 
-    List<UserData> list = new ArrayList<>();
+    List<UserData> userList = new ArrayList<>();
 
-    list.add(UserData[0]);
-    list.add(UserData[1]);
+    userList.add(userData[0]);
+    userList.add(userData[1]);
 
-    return list;
+    return userList;
   }
 
+  @GetMapping("/names/{user-birthday}/{user-name}")
```

Comment:
生年月日を基準にユーザー名を指定してリソースが特定できるならこのURLでもいいのですが、多分違うと思うので、
```suggestion
  @GetMapping("/names/{user-birthday}/{user-name}")
```

`http://localhost:8080/names?birthday=2000/01/01&name=koyama`
のようにクエリ文字列を受け取る形にするのはどうでしょうか？

クエリ文字列は検索用途によく使います。

実装方法は`@RequestParam`で検索してみてください！

Reply:
そのように考えてクエリ文字列と使い分けをしていらっしゃるのですね。
色々とコードを書くうちに自分もそのような考え方が見えてくる事を願います。
ご指摘ありがとうございます！